misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,300 @@ use strict;
+            $transcript_result->{$transcript->stable_id} = $score * $PENALTY;
+            $tl_transcript_result->{$transcript->stable_id} = $tl_score * $PENALTY;
           }
         }
 
-        my $best_score = 0;
-        my $best_tl_score = 0;
-        my $best_id;
-        my ($score, $tl_score);
-# Comparing the scores based on coding exon overlap
-# If there is a stale mate, chose best exon overlap score
-        foreach my $tid (sort { $transcript_result{$b} <=> $transcript_result{$a} } keys(%transcript_result)) {
-          $score = $transcript_result{$tid};
-          $tl_score = $tl_transcript_result{$tid};
-          if ($score > $transcript_score_threshold || $tl_score > $tl_transcript_score_threshold) {
-            if ($tl_score >= $best_tl_score) {
-              if ($tl_score > $best_tl_score) {
-                $best_id = $tid;
-                $best_score = $score;
-                $best_tl_score = $tl_score;
-              } elsif ($tl_score == $best_tl_score) {
-                if ($score > $best_score) {
-                  $best_id = $tid;
-                  $best_score = $score;
-                }
-              }
-            } elsif ($score >= $best_score) {
-              $best_id = $tid;
-              $best_score = $score;
-            }
-          }
-        }
+        my ($best_id, $best_score, $best_tl_score) = $self->compute_best_scores($transcript_result, $tl_transcript_result);
 
-# If a best match was defined for the refseq transcript, store it as direct xref for ensembl transcript
+        # If a best match was defined for the refseq transcript, store it as direct xref for ensembl transcript
         if ($best_id) {
-          my ($acc, $version) = split(/\./, $id);
-	  $version =~ s/\D//g if $version;
-          my $source_id;
-          $source_id = $mrna_source_id if $acc =~ /^NM_/;
-          $source_id = $ncrna_source_id if $acc =~ /^NR_/;
-          $source_id = $pred_mrna_source_id if $acc =~ /^XM_/;
-          $source_id = $pred_ncrna_source_id if $acc =~ /^XR_/;
-          # Accession should be of format NM_/XM_/NR_/XR_ otherwise it is not valid
-          if (!defined $source_id) { next; }
-          my $xref_id = $self->add_xref({ acc => $acc,
-                                          version => $version,
-                                          label => $id,
-                                          desc => undef,
-                                          source_id => $source_id,
-                                          species_id => $species_id,
-                                          dbi => $dbi,
-                                          info_type => 'DIRECT' });
-          $self->add_direct_xref($xref_id, $best_id, "Transcript", "", $dbi);
-
-	  my $t_of = $transcript_of;
-          my $g_of = $t_of->get_Gene();
-          my $entrez_id = $g_of->stable_id();
-          my $tl_of = $t_of->translation();
+          my ($acc, $version) = split(/\./xms, $id);
```

Comment:
I think perhaps observing the fully annotate regex options for 'split on period' is going overboard.